internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+	diags.Append(respDiags...)
+	state.TokenExchangeProcessorPolicyRef = tokenExchangeProcessorPolicyRefToState
+
+	state.RefreshTokenRollingGracePeriodType = types.StringPointerValue(r.RefreshTokenRollingGracePeriodType)
+	state.RefreshTokenRollingGracePeriod = types.Int64PointerValue(r.RefreshTokenRollingGracePeriod)
+	state.ClientSecretRetentionPeriodType = types.StringPointerValue(r.ClientSecretRetentionPeriodType)
+	state.ClientSecretRetentionPeriod = types.Int64PointerValue(r.ClientSecretRetentionPeriod)
+	state.ClientSecretChangedTime = types.StringValue(r.GetClientSecretChangedTime().Format(time.RFC3339Nano))
+	state.TokenIntrospectionSigningAlgorithm = types.StringPointerValue(r.TokenIntrospectionSigningAlgorithm)
+	state.TokenIntrospectionEncryptionAlgorithm = types.StringPointerValue(r.TokenIntrospectionEncryptionAlgorithm)
+	state.TokenIntrospectionContentEncryptionAlgorithm = types.StringPointerValue(r.TokenIntrospectionContentEncryptionAlgorithm)
+	state.JwtSecuredAuthorizationResponseModeSigningAlgorithm = types.StringPointerValue(r.JwtSecuredAuthorizationResponseModeSigningAlgorithm)
+	state.JwtSecuredAuthorizationResponseModeEncryptionAlgorithm = types.StringPointerValue(r.JwtSecuredAuthorizationResponseModeEncryptionAlgorithm)
+	state.JwtSecuredAuthorizationResponseModeContentEncryptionAlgorithm = types.StringPointerValue(r.JwtSecuredAuthorizationResponseModeContentEncryptionAlgorithm)
+
+	return diags
+}
+
+func grantTypes(grantTypesSet types.Set) []string {
+	var grantTypesSlice []string
+	for _, grantType := range grantTypesSet.Elements() {
+		grantTypesSlice = append(grantTypesSlice, grantType.(types.String).ValueString())
+	}
+	return grantTypesSlice
+}
+
+func addOptionalOauthClientFields(ctx context.Context, addRequest *client.Client, plan oauthClientResourceModel) error {
+
+	if internaltypes.IsDefined(plan.ExclusiveScopes) {
+		var slice []string
+		plan.ExclusiveScopes.ElementsAs(ctx, &slice, false)
+		addRequest.ExclusiveScopes = slice
+	}
+
+	if internaltypes.IsDefined(plan.Enabled) {
+		addRequest.Enabled = plan.Enabled.ValueBoolPointer()
+	}
+
+	if internaltypes.IsDefined(plan.RedirectUris) {
+		var slice []string
+		plan.RedirectUris.ElementsAs(ctx, &slice, false)
+		addRequest.RedirectUris = slice
+	}
+
+	if internaltypes.IsDefined(plan.Description) {
+		addRequest.Description = plan.Description.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.LogoUrl) {
+		addRequest.LogoUrl = plan.LogoUrl.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.DefaultAccessTokenManagerRef) {
+		addRequest.DefaultAccessTokenManagerRef = client.NewResourceLinkWithDefaults()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.DefaultAccessTokenManagerRef, false)), addRequest.DefaultAccessTokenManagerRef)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.RestrictToDefaultAccessTokenManager) {
```

Comment:
Not 100% sure but I don't think the IsDefined checks are necessary when we aren't using the Json Unmarshal method. If it's not defined, the ValueXPointer method should return nil, and that's what we'd want.

Reply:
these are all updated besides sets and objects